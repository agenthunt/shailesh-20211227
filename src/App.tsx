/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * Generated with the TypeScript template
 * https://github.com/react-native-community/react-native-template-typescript
 *
 * @format
 */

import {useAppState} from '@react-native-community/hooks';
import React, {useEffect, useRef, useState} from 'react';
import {
  AppState,
  Button,
  Platform,
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import {useAppDimensions, useAppDispatch} from './app/hooks';
import {Orderbook} from './features/orderbook/orderbook';
import {orderbookSlice} from './features/orderbook/orderbookSlice';

const App = () => {
  const appState = useRef(AppState.currentState);
  const [appStateVisible, setAppStateVisible] = useState(appState.current);
  const dispatch = useAppDispatch();
  const {isDesktop} = useAppDimensions();
  useEffect(() => {
    const subscription = AppState.addEventListener('change', nextAppState => {
      if (
        appState.current.match(/active/) &&
        nextAppState.match(/inactive|background/)
      ) {
        console.log('App is going to background!');
        dispatch(orderbookSlice.actions.unsubscribeToOrderbook());
      }

      appState.current = nextAppState;
      setAppStateVisible(appState.current);
      console.log('AppState', appState.current);
    });

    return () => {
      subscription.remove();
    };
  }, []);
  return (
    <SafeAreaView data-cy="safeAreaView" style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <Orderbook />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    height: Platform.OS === 'web' ? '100vh' : '100%',
    justifyContent: 'flex-start',
    alignItems: 'stretch',
  },
  scrollView: {},
  contentContainerStyle: {
    height: Platform.OS === 'web' ? '100vh' : '100%',
  },
});

export default App;
