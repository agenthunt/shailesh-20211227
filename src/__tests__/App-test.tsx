/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('App component tests', () => {
  const initialState = {
    orderbook: {
      asks: [],
      bids: [],
    },
  };
  const mockStore = configureStore();
  let store;

  it('renders default state of App.jsx correctly', () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
