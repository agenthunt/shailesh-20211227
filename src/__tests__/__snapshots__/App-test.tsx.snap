// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests renders default state of App.jsx correctly 1`] = `
<RCTSafeAreaView
  data-cy="safeAreaView"
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "stretch",
      "flex": 1,
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#1c2a31",
          "flex": 1,
          "padding": 8,
        },
        Object {
          "height": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.1)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <ActivityIndicator
        size="large"
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
            "fontSize": 26,
            "paddingBottom": 8,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
         Order Book 
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#fff",
            "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
            "fontSize": 12,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginBottom": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.33,
                "paddingRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "borderBottomColor": "#98a6af",
                  "borderBottomWidth": 0.3,
                  "borderRadius": 1,
                  "borderStyle": "dashed",
                  "color": "#98a6af",
                  "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
                  "fontSize": 12,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              PRICE
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.33,
                "paddingRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "borderBottomColor": "#98a6af",
                  "borderBottomWidth": 0.3,
                  "borderRadius": 1,
                  "borderStyle": "dashed",
                  "color": "#98a6af",
                  "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
                  "fontSize": 12,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              SIZE
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-end",
                "flex": 0.33,
                "paddingRight": 20,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "flex-end",
                  "borderBottomColor": "#98a6af",
                  "borderBottomWidth": 0.3,
                  "borderRadius": 1,
                  "borderStyle": "dashed",
                  "color": "#98a6af",
                  "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
                  "fontSize": 12,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              TOTAL
            </Text>
          </View>
        </View>
        <RCTScrollView
          data={Array []}
          getItem={[Function]}
          getItemCount={[Function]}
          getItemLayout={[Function]}
          invertStickyHeaders={true}
          inverted={true}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          style={
            Array [
              Object {
                "transform": Array [
                  Object {
                    "scaleY": -1,
                  },
                ],
              },
              undefined,
            ]
          }
          viewabilityConfigCallbackPairs={Array []}
        >
          <View />
        </RCTScrollView>
      </View>
      <View
        style={
          Array [
            Object {
              "padding": 8,
            },
            Object {
              "alignSelf": "center",
              "justifyContent": "center",
              "margin": "auto",
            },
          ]
        }
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RCTScrollView
          data={Array []}
          getItem={[Function]}
          getItemCount={[Function]}
          getItemLayout={[Function]}
          invertStickyHeaders={false}
          inverted={false}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          stickyHeaderIndices={Array []}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
    <View
      style={
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "padding": 8,
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
                  "fontSize": 12,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              Subscribe
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "margin": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
                "padding": 8,
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "LED SCREEN GTAMBLOG CAPS+ 2.0",
                  "fontSize": 12,
                  "textAlign": "center",
                  "textAlignVertical": "center",
                }
              }
            >
              Toggle Feed
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
